src/components/MoveAdder.ts

```diff
@@ -15,4 +15,247 @@ export class MoveAdder<TGameStartr extends FullScreenPokemon> extends GeneralCom
+                        },
+                        {
+                            text: moves[1].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 1),
+                        },
+                        {
+                            text: moves[2].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 2),
+                        },
+                        {
+                            text: moves[3].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 3),
+                        }],
+                });
+                this.gameStarter.menuGrapher.setActiveMenu("BattleFightList");
+            },
+        );
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+    /**
+     * A function that gives the final dialog lines for resolving a move conflict as well as actually teaches
+     * a Pokemon a new move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    private teachMove(pokemon: IPokemon, move: IMove, index: number) {
+        this.gameStarter.menuGrapher.createMenu("GeneralText", {
+            deleteOnFinish: true,
+        });
+        this.gameStarter.menuGrapher.addMenuDialog(
+            "GeneralText",
+            [
+                [
+                    "1, 2 and... Poof!",
+                ],
+                pokemon.title.join("") + " forgot " + pokemon.moves[index].title.toUpperCase() + "!",
+                [
+                    "And...",
+                ],
+                pokemon.title.join("") + " learned " + move.title.toUpperCase() + "!",
+            ],
+            (): void => {
+                this.addMove(pokemon, move, index);
+        });
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+    /**
+     * Brings up the dialog for when a move conflict exists and the player doesn't want to continue.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    private refuseLearnMove(pokemon: IPokemon, move: IMove) {
+        this.gameStarter.menuGrapher.deleteMenu("Yes/No");
+        this.gameStarter.menuGrapher.deleteMenu("GeneralText"); //it's needed
+        this.gameStarter.menuGrapher.createMenu("GeneralText"); //you have to link back to lol() if it's no.
+
+        this.gameStarter.menuGrapher.addMenuDialog(
+            "GeneralText",
+            [
+                [
+                    "Abandon learning " + move.title.toUpperCase() + "?",
+                ],
+            ],
+            (): void => {
+                this.gameStarter.menuGrapher.createMenu("Yes/No", {
+                    killOnB: ["GeneralText"], //kills menu when this menu is killed
```

Comment:
//kills menu when this menu is killed

Implied by the code. Remove in all places this shows up.